libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SendActivity.cs

```diff
@@ -113,6 +114,15 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             return await dc.EndDialogAsync(response, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Add context to activity.
+        /// </summary>
+        /// <param name="dc">Dialog context.</param>
+        /// <param name="activity">Activity to modify.</param>
+        protected virtual void AddContext(DialogContext dc, Activity activity)
+        { 
+        }
```

Comment:
Wha?  Why?

Reply:
Because Ask derives from this and I needed a way to add the context.  The other alternative was to duplicate BeginDialogAsync or pull all of it out into a common method.  Do you have a preference?